fsm.go

```diff
@@ -237,3 +249,7 @@ func (r *fsmS) cpuSetFileContent(pid int) string {
 
 	return string(data)
 }
+
+func getRetryPeriodForIteration(retryNumber int) time.Duration {
```

Comment:
This function is very specific as suggested by its long name 😅 How about we allow callers to provide the initial delay as well? Then we could name it simply `expDelay()` or something like this. WDYT?

Reply:
`expDelay` sounds good